### PR TITLE
Add google protos to buildbucket for easier creation of buildbucket objects

### PR DESCRIPTION
**Pull request by ricardoamador**
When adding properties to the build bucket requests for buildbucket_v2 we need some of the data types from the google base protobufs. 

*List which issues are fixed by this PR. You must list at least one issue.*
Contributes to https://github.com/flutter/flutter/issues/135934

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/buildbucket-dart/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# 1.0.7
+
+- Expose the google.protobuf dependency ojects.
```
Comment (yusuf-goog) [inline packages/buildbucket-dart/CHANGELOG.md:3]: typo

File: packages/buildbucket-dart/lib/src/generated/go.chromium.org/luci/buildbucket/proto/build.pbenum.dart [hunk 2]
```diff
@@ -14,20 +14,14 @@ import 'dart:core' as $core;
...
-      BuildInfra_Buildbucket_ExperimentReason._(2, _omitEnumNames ? '' : 'EXPERIMENT_REASON_BUILDER_CONFIG');
-  static const BuildInfra_Buildbucket_ExperimentReason EXPERIMENT_REASON_GLOBAL_MINIMUM =
-      BuildInfra_Buildbucket_ExperimentReason._(3, _omitEnumNames ? '' : 'EXPERIMENT_REASON_GLOBAL_MINIMUM');
-  static const BuildInfra_Buildbucket_ExperimentReason EXPERIMENT_REASON_REQUESTED =
-      BuildInfra_Buildbucket_ExperimentReason._(4, _omitEnumNames ? '' : 'EXPERIMENT_REASON_REQUESTED');
-  static const BuildInfra_Buildbucket_ExperimentReason EXPERIMENT_REASON_GLOBAL_INACTIVE =
-      BuildInfra_Buildbucket_ExperimentReason._(5, _omitEnumNames ? '' : 'EXPERIMENT_REASON_GLOBAL_INACTIVE');
-
-  static const $core.List<BuildInfra_Buildbucket_ExperimentReason> values = <BuildInfra_Buildbucket_ExperimentReason>[
+  static const BuildInfra_Buildbucket_ExperimentReason EXPERIMENT_REASON_UNSET = BuildInfra_Buildbucket_ExperimentReason._(0, _omitEnumNames ? '' : 'EXPERIMENT_REASON_UNSET');
```
Comment (yusuf-goog) [inline packages/buildbucket-dart/lib/src/generated/go.chromium.org/luci/buildbucket/proto/build.pbenum.dart:17]: Does the line limit match style?

Comment (ricardoamador) [reply packages/buildbucket-dart/lib/src/generated/go.chromium.org/luci/buildbucket/proto/build.pbenum.dart]: Good catch. I forgot to run the formatter but looks like we ignore these files when linting?

File: packages/buildbucket-dart/CHANGELOG.md (hunk 1 above)
Comment (ricardoamador) [reply packages/buildbucket-dart/CHANGELOG.md]: Thank you!

**Review by keyonghan — COMMENTED** [review]
Are these all auto generated files? How do we generate them?

**Comment by ricardoamador** [comment]
In the regenerate.sh script we check out the repo and build them as they are dependencies for the build bucket protos and objects created there. The underlying issue is that properties are a Struct, which is a Map<String, Value> and Value is a proto message that has types in it. 

You know what, hold that thought. I might not need this change then.

**Comment by ricardoamador** [comment]
Okay I do need this. For whatever reason the properties message is read only so if I get the fields object from it I need the Struct and or Value object to be able to cast or do anything with it.

**Review by keyonghan — APPROVED** [review]
LGTM

**Pull request merged**